org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/TextEditAcceptor.java

```diff
@@ -0,0 +1,74 @@
+/*******************************************************************************
+ * Copyright (c) 2022 itemis AG (http://www.itemis.eu) and others.
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+package org.eclipse.xtext.ide.server;
+
+/**
+ * @author Rub�n Porras Campo - Initial contribution and API, factored out of ChangeConverter2
+ *
+ * @since 2.27
+ */
+import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import org.eclipse.lsp4j.ClientCapabilities;
+import org.eclipse.lsp4j.InitializeParams;
+import org.eclipse.lsp4j.TextDocumentEdit;
+import org.eclipse.lsp4j.TextEdit;
+import org.eclipse.lsp4j.VersionedTextDocumentIdentifier;
+import org.eclipse.lsp4j.WorkspaceClientCapabilities;
+import org.eclipse.lsp4j.WorkspaceEdit;
+import org.eclipse.lsp4j.WorkspaceEditCapabilities;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+
+public class TextEditAcceptor {
+	private final boolean useDocumentChanges;
+	private final WorkspaceEdit edit;
+
+	public TextEditAcceptor(WorkspaceEdit edit, ILanguageServerAccess access) {
+		this.edit = edit;
+
+		Boolean documentChanges = null;
+		if (access != null) {
+			InitializeParams initializeParams = access.getInitializeParams();
+			if (initializeParams != null) {
+				ClientCapabilities clientCapabilities = initializeParams.getCapabilities();
+				if (clientCapabilities != null) {
+					WorkspaceClientCapabilities workspaceClientCapabilities = clientCapabilities.getWorkspace();
+					if (workspaceClientCapabilities != null) {
+						WorkspaceEditCapabilities workspaceEditCapabilities = workspaceClientCapabilities.getWorkspaceEdit();
+						if (workspaceEditCapabilities != null) {
+							documentChanges = workspaceEditCapabilities.getDocumentChanges();
+						}
+					}
+				}
+			}
+		}
+		this.useDocumentChanges = documentChanges == Boolean.TRUE;
+		if (this.useDocumentChanges) {
+			this.edit.setDocumentChanges(new ArrayList<>());
+		} else {
+			this.edit.setChanges(new LinkedHashMap<>());
+		}
+	}
+
+	public Object accept(String theUri, Document document, List<TextEdit> textEdits) {
```

Comment:
This should be a void method as most of the Acceptor-style APIs in Xtext are?

Currently it either return `true` or a random old textEdits or null, none of these do make sense to me.

Reply:
done